robot/client/client_test.go

```diff
@@ -332,15 +331,10 @@ func TestStatusClient(t *testing.T) {
 	var imgBuf bytes.Buffer
 	test.That(t, png.Encode(&imgBuf, img), test.ShouldBeNil)
 
-	var imageReleased bool
-	var imageReleasedMu sync.Mutex
-	injectCamera.StreamFunc = func(ctx context.Context, errHandlers ...gostream.ErrorHandler) (gostream.VideoStream, error) {
-		return gostream.NewEmbeddedVideoStreamFromReader(gostream.VideoReaderFunc(func(ctx context.Context) (image.Image, func(), error) {
-			imageReleasedMu.Lock()
-			imageReleased = true
-			imageReleasedMu.Unlock()
-			return img, func() {}, nil
-		})), nil
+	injectCamera.ImageFunc = func(ctx context.Context, mimeType string, extra map[string]interface{}) ([]byte, camera.ImageMetadata, error) {
+		resBytes, err := rimage.EncodeImage(ctx, img, mimeType)
+		test.That(t, err, test.ShouldBeNil)
+		return resBytes, camera.ImageMetadata{MimeType: mimeType}, nil
```

Comment:
Why do we need to encode here? Can't we just return `imgBuf.Bytes(), camera.ImageMetadata{MimeType: "whatever a png is"}, nil`

Reply:
Done